services/filemanager/gcsmanager.go

```diff
@@ -66,9 +66,6 @@ func (manager *GCSManager) ListFilesWithPrefix(ctx context.Context, startAfter,
 		return
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, manager.getTimeout())
-	defer cancel()
-
 	// Create GCS Bucket handle
 	if manager.Config.Iterator == nil {
```

Comment:
@BonapartePC why are we caching the iterator? Is this thread safe??

Reply:
The way this function is being used is: With a given manager, calling subsequent ListFilesWithPrefix() should return us continuous items one after the other. This is the reason we cache it in the manager itself.

Reply:
This is not safe, we need to revisit. Can we guarantee that filemanager will be used exclusively by one goroutine and then never reused?

Reply:
Currently, all of the code is written keeping that in mind - https://github.com/rudderlabs/rudder-server/blob/master/services/filemanager/s3manager.go#L207

Reply:
this is a bad api design, I am confident we can do much better than that